lib/CL/devices/pthread/pthread_scheduler.c

```diff
@@ -290,14 +289,19 @@ work_group_scheduler (kernel_run_command *k,
   assert (pc.printf_buffer_position != NULL);
 
   /* Flush to zero is only set once at start of kernel (because FTZ is
-   * a compilation option), but we need to reset rounding mode after every
-   * iteration (since it can be changed during kernel execution). */
+   * a compilation option) */
   unsigned flush = k->kernel->program->flush_denorms;
-  if (thread_data->current_ftz != flush)
-    {
-      pocl_set_ftz (flush);
-      thread_data->current_ftz = flush;
-    }
+  cl_device_fp_config supports_any_denorms
```

Comment:
Can we refactor the FP setup to a function which both basic and pthread can use?